### PR TITLE
add dial and recvline

### PR DESCRIPTION
**Pull request by hopkiw**

### CODE REVIEW

File: src/nss/new_nss_oslogin.c [hunk 1]
```diff
@@ -157,3 +167,76 @@ int parsegroup(char *str, struct group *result, char *buffer, size_t buflen) {
...
+      buffer->bufsize = new_size;
+    }
+
+    memcpy(&(buffer->buf[buffer->buflen]), recvbuf, recvlen);
+    buffer->buflen += recvlen;
+
+    if (recvbuf[recvlen - 1] == '\n') {
+      return buffer->buflen;
+    }
+  }
```
Comment (winterfreshx) [inline src/nss/new_nss_oslogin.c:241]: Explicit return value at the end

Comment (hopkiw) [reply src/nss/new_nss_oslogin.c:241]: this is within a while loop

File: src/nss/new_nss_oslogin.c [hunk 2]
```diff
@@ -157,3 +167,76 @@ int parsegroup(char *str, struct group *result, char *buffer, size_t buflen) {
...
+
+struct Buffer {
+  ssize_t buflen; // how much data we read into the buffer
+  ssize_t bufsize; // allocated space for buffer
+  char *buf;  // the buffer we copy results into
+  int socket;
+};
+
+struct Buffer pwbuf;
+struct Buffer grbuf;
```
[the remark this answers is not in the document]
Comment (hopkiw) [reply src/nss/new_nss_oslogin.c:179]: these are the global buffers used for the `getpwent` and `getgrent` functions, not yet implemented

File: src/nss/new_nss_oslogin.c (hunk 1 above)
Comment (winterfreshx) [reply src/nss/new_nss_oslogin.c:241]: It's for the end of the function, there is no explicit return value. Similar to how you return 0 for dial.

File: src/nss/new_nss_oslogin.c [hunk 3]
```diff
@@ -157,3 +167,76 @@ int parsegroup(char *str, struct group *result, char *buffer, size_t buflen) {
...
+  strcpy(remote.sun_path, SOCK_PATH);
+  len = strlen(remote.sun_path) + sizeof(remote.sun_family);
+  if (connect(buf->socket, (struct sockaddr *)&remote, len) == -1) {
+      return -1;
+  }
+
+  return 0;
+}
+
+int recvline(struct Buffer *buffer) {
```
Comment (winterfreshx) [inline src/nss/new_nss_oslogin.c:201]: Make it a const since we're not changing pointer

File: src/nss/new_nss_oslogin.c [hunk 4]
```diff
@@ -157,3 +167,76 @@ int parsegroup(char *str, struct group *result, char *buffer, size_t buflen) {
...
+  ssize_t buflen; // how much data we read into the buffer
+  ssize_t bufsize; // allocated space for buffer
+  char *buf;  // the buffer we copy results into
+  int socket;
+};
+
+struct Buffer pwbuf;
+struct Buffer grbuf;
+
+int dial(struct Buffer *buf) {
```
Comment (winterfreshx) [inline src/nss/new_nss_oslogin.c:181]: Make it a const since we're not changing pointer

**Review by winterfreshx — COMMENTED** [review]
Just a few small nits, I am ok with signing off but will wait for response to comments before I push the button =)

**Review by winterfreshx — APPROVED** [review]

**Pull request merged**